### ISSUE TITLE
How to set OpenTelemetry tracing parameters?

### ISSUE THREAD

**Issue by boramalper** [cloudevents/sdk-go#693]
I am using [Knative Eventing](https://knative.dev/docs/eventing/) and want to set [OpenTelemetry](https://opentelemetry.io/) tracing parameters (tracing ID, span ID etc.) for the event that I am sending to [Knative Eventing Broker](https://knative.dev/docs/eventing/broker/).

I thought `WithTracePropagation()` could have been helpful but it is [deprecated](https://pkg.go.dev/github.com/cloudevents/sdk-go/v2@v2.4.1/client#WithTracePropagation) and no alternatives are listed. [`ObservabilityService`](https://pkg.go.dev/github.com/cloudevents/sdk-go/v2@v2.4.1/client#ObservabilityService) could have been an alternative solution but it does not have an `OutboundContextDecorators()`... Though I would like to be able to set OpenTelemetry tracing parameters myself freely, without using middlewares or other magic.

What I currently have is this (simplified):
```go
import(
	cloudevents "github.com/cloudevents/sdk-go/v2"
	"github.com/google/uuid"
)


type MyStruct struct {
	// ...
}

func main() {	
	event := cloudevents.NewEvent("1.0")
	event.SetID(uuid.New().String())
	event.SetType("my-type")
	event.SetSource("//my-source")

	_ = event.SetData(cloudevents.ApplicationJSON, MyStruct{})

	ceClient, _ := cloudevents.NewClientHTTP()
	// I want to set OpenTelemetry trace ID here!
	if result := ceClient.Send(cloudevents.ContextWithTarget(ctx, env.Sink), event); !cloudevents.IsACK(result) {
		log.Fatalf("failed to send CloudEvent: %+v", result)
	}
}
```

**Comment by boramalper** [comment]
@ustiugov

**Comment by ustiugov** [comment]
hi @joaopgrassi, this is definitely something we would be interested in. Would be great if you could upstream that. 

thanks @boramalper for bringing me into this discussion.

**Comment by joaopgrassi** [comment]
Hi @ustiugov cool, I'll working on it. I feel though there's some scattered issues about this. I found this old one https://github.com/cloudevents/sdk-go/issues/554 and commented there about the sdk spec about this observability service.

**Comment by n3wscott** [comment]
Yes please, glad you got it working. @benmoss was also helping clear some of this up with some examples added to the sdk. Maybe Ben has more to say?

**Comment by benmoss** [comment]
With the changes I made in #708 you can propagate trace headers through with the [otelhttp middleware](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#Handler), I did this in this demo app: https://github.com/benmoss/knative-tracing/blob/main/cmd/coinflip/main.go#L60-L62

I think this is nice from a simplicity perspective, but having an ObservabilityService implementation seems like a good idea too.

**Comment by ustiugov** [comment]
@benmoss thanks a lot for your response! this seems very useful and easy to use. do you know what is the minimal version of otelhttp that supports it?

**Comment by benmoss** [comment]
@ustiugov it looks like it's been in since 0.12.0 https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.12.0
